src/java.base/share/classes/java/net/Socket.java

```diff
@@ -616,10 +613,15 @@ void setConnected() {
      *        cause it to wakeup and close the socket. This method will then throw
      *        {@code SocketException} with the interrupt status set.
      * </ol>
+     * <p> Besides above noted cases, the socket will also be closed when a
+     * connection attempt fails with an {@link IOException}.
      *
      * @param   endpoint the {@code SocketAddress}
      * @throws  IOException if an error occurs during the connection, the socket
      *          is already connected or the socket is closed
+     * @throws  UnknownHostException if the endpoint address is
+     *          {@linkplain InetSocketAddress#isUnresolved() not resolved},
```

Comment:
I think better to say "if the endpoint is an unresolved InetSocketAddress".

Reply:
Fixed in 7e633d8d213a950304c550f93a1697c1f2530a05.